gradle build support

takaczapka opened the pull request:
Hey,

I've added a gradle wrapper into the project so now you can create IntelliJ or Eclipse project with following:
> cd koans

> ./gradlew idea
or
> ./gradle eclipse

BTW I really like the project and I'm planning to use it during Code Club classes we're running for kids - I'll probably simplify, remove and extend a few koan sets to help 10-12 year olds slowly build a comprehension of JAVA. My end goal is to show them how to hack Minecraft mods, but without good JAVA foundations we can't go straight into it.

matyb commented:
Thanks @takaczapka big changes, look safe, and they seem to simplify things  greatly for those using IDEs. Interesting you reached for the maven folder structure, I did the same thing - only to get squeemish. It's a fine line between idiomatic java (ie maven folder structure) and something a tad too prescriptive or that would give the impression of contrived complexity. 

Appreciate the PR. Thanks again.

takaczapka commented:
Hey Mat, 

I've done today a few more things, but I'm not entirely sure if this is something you would like to take on into the main project branch.

Main things are:
* added a few more koans (in the beginner's section)
* changed a sequence of some koans
* converted lib/ project to gradle project by
    * merging all sources into one folder  (src/main/java, src/main/test)
    * all tests can be run from either IDE or command line (I've spent a while on this one)
* introduced a flag in config.properties: enable_expectation_result - the flag when false will make a koans runtime not to show an expected value info. Why? I think if the user knows what is expected, then she might be tempted to copy/paste the result to the source code. So by default I don't wanna show it. If I remember properly Scala/Ruby koans don't show anything like this.

If you want me I can send you a pull request. It's up to you.

matyb commented:
@takaczapka definitely interested, more koans is almost always a good thing and the sequence I'm never really sure about (hard to think of a progression that fits everyone).

it certainly couldn't hurt, worst case I'd probably merge it in and tweak something. But I certainly like the notes, each feature sounds genuinely helpful and I like your style.
